utils/symbolicate-linux-fatal

```diff
@@ -0,0 +1,162 @@
+        if dynlib_fname in dyn_libs:
+            dynlib_path = dyn_libs[dynlib_fname]
+        elif dynlib_fname in binary:
+            dynlib_path = binary
+        else:
+            dynlib_path = None
+
+        if "<unavailable>" in stack_tokens[3]:
+            framePC = int(stack_tokens[2], 16)
+            symbol_offset = int(stack_tokens[-1], 10)
+            dynlib_baseaddr = framePC - symbol_offset
+            if dynlib_path in memmap:
+                if memmap[dynlib_path] != dynlib_baseaddr:
+                    error_msg = "Mismatched base address for: {0:s}, " \
+                                "had: {1:x}, now got {2:x}"
+                    error_msg = error_msg.format(
+                        dynlib_path, memmap[dynlib_path], dynlib_baseaddr)
+                    raise Exception(error_msg)
+            else:
+                memmap[dynlib_path] = dynlib_baseaddr
+        else:
+            framePC = int(stack_tokens[2], 16) + int(stack_tokens[-1], 10)
+        full_stack.append(
+            {"line": line, "framePC": framePC, "dynlib_fname": dynlib_fname})
+
+    lldb_target = create_lldb_target(binary, memmap)
+    frame_idx = 0
+    for frame in full_stack:
+        use_orig_line = True
+        frame_addr = frame["framePC"]
+        dynlib_fname = frame["dynlib_fname"]
+        so_addr = lldb_target.ResolveLoadAddress(frame_addr-1)
+        sym_ctx = so_addr.GetSymbolContext(lldb.eSymbolContextEverything)
+        frame_fragment = "{0: <4d} {1:20s} 0x{2:016x}".format(
+            frame_idx, dynlib_fname, frame_addr)
+        symbol = sym_ctx.GetSymbol()
+        if symbol.IsValid():
+            symbol_base = symbol.GetStartAddress().GetLoadAddress(lldb_target)
+            symbol_fragment = "{0:s} + {1:d}".format(
+                symbol.GetName(), frame_addr - symbol_base)
+            use_orig_line = False
+        else:
+            symbol_fragment = "<unavailable>"
+        line_entry = sym_ctx.GetLineEntry()
+        if line_entry.IsValid():
+            line_fragment = "at {0:s}:{1:d}".format(
+                line_entry.GetFileSpec().GetFilename(), line_entry.GetLine())
+        else:
+            line_fragment = ""
+        if use_orig_line:
+            print(frame["line"].rstrip())
+        else:
+            print("{0:s}    {1:s} {2:s}".format(
+                frame_fragment, symbol_fragment, line_fragment))
+        frame_idx = frame_idx + 1
+
+
+def main():
+    parser = argparse.ArgumentParser(
+        formatter_class=argparse.RawDescriptionHelpFormatter,
+        description="""Symbolicates stack traces in Linux log files.""")
+    parser.add_argument(
+        "binary", help="Executable which produced the log file")
+    parser.add_argument(
+        "log", help="Log file containing the stack trace to symbolicate")
+    args = parser.parse_args()
+
+    binary = args.binary
+    if args.log == "-":
+        logfile = sys.stdin
+    else:
+        logfile = open(args.log, 'rU')
```

Comment:
Just a Python `argparse` tip:
```
parser.add_argument(
    "log", type=argparse.FileType("rU"), ...)
```
is the shorthand for this.
see https://docs.python.org/2/library/argparse.html#filetype-objects